src/luarocks/cmd/new_version.lua

```diff
@@ -48,6 +48,7 @@ if it already exists.]], util.see_also())
 
    cmd:option("--dir", "Output directory for the new rockspec.")
    cmd:option("--tag", "New SCM tag.")
+   cmd:flag("--keep-url", "Keep the old url if a new one is not specified.")
```

Comment:
Perhaps invert the flag?
i.e. have a `--guess-url` that defaults to `true` and then you can do `--guess-url=false`

Reply:
I am not particularly fond of that style, it sort of pollutes the command line. What about `--no-guess-url`  ?